src/Sarif.Viewer.VisualStudio.Core/Services/DataService.cs

```diff
@@ -80,14 +103,27 @@ private async Task SendEnhancedResultDataAsync(SarifLog sarifLog)
                     dataCache.SarifErrors.Add(sarifErrorListItem);
                 }
 
-                ISarifErrorListEventSelectionService sarifErrorListEventSelectionService = componentModel.GetService<ISarifErrorListEventSelectionService>();
-                sarifErrorListEventSelectionService.NavigatedItem = items[0];
-                sarifErrorListEventSelectionService.SelectedItem = items[0];
+                this.sarifErrorListEventSelectionService.NavigatedItem = items[0];
+                this.sarifErrorListEventSelectionService.SelectedItem = items[0];
 
                 items[0].Locations?.FirstOrDefault()?.NavigateTo(usePreviewPane: false, moveFocusToCaretLocation: true);
             }
 
             SarifExplorerWindow.Find()?.Show();
+
+            return runIndex;
+        }
+
+        private async Task CloseEnhancedResultDataAsync(int cookie)
```

Comment:
actually we don't need the cookie to clean up since we only keep current key event data.
if we will support multiple key event data, the cookie can be used to distinguish them.